tools/matRad_fitBaseData.m

```diff
@@ -80,49 +83,85 @@
-                             
-                                     
-    options1.lb =   0;
-    options1.ub = Inf;
-    options1.ipopt.hessian_approximation = 'limited-memory';
-    options1.ipopt.limited_memory_update_type = 'bfgs';
-    options1.ipopt.print_level = 1;
-    start1 = 8;
-    
-    [fitResult1, ~] = ipopt (start1, funcs1, options1);                   
-    preSigma1 = fitResult1;   
-    
-    %define fit parameters
-    funcs2.objective = @(p) sum(sum((gauss2(axisGauss, p(1), p(2), p(3)) - profile).^2));
-
-    funcs2.gradient = @(p) [ 2 * sum(sum((gauss2(axisGauss, p(1), p(2), p(3)) - profile) .* ...  
-                                    (-1 / (2 * pi * p(2)^2) .* exp(-axisGauss.^2 / (2 * p(2)^2)) + 1 / (2 * pi * p(3)^2) .* exp(-axisGauss.^2 / (2 * p(3)^2)))));
-                             2 * sum(sum((gauss2(axisGauss, p(1), p(2), p(3)) - profile) .* ...                                   
-                                    (1 - p(1)) .* exp(-axisGauss.^2 / (2 * p(2)^2)) .* (axisGauss.^2 - 2 * p(2)^2) / (2 * pi * p(2)^5))); 
-                             2 * sum(sum((gauss2(axisGauss, p(1), p(2), p(3)) - profile) .* ...
-                                    p(1)       .* exp(-axisGauss.^2 / (2 * p(3)^2)) .* (axisGauss.^2 - 2 * p(3)^2) / (2 * pi * p(3)^5)))]; 
-
-    options2.lb = [   0,   preSigma1 - 1,  preSigma1 + 1];
-    options2.ub = [ 0.2,   preSigma1 + 1,  preSigma2 + 10];
-%     options.ipopt.tol = 1e-30;
-
-    options2.ipopt.hessian_approximation = 'limited-memory';
-    options2.ipopt.limited_memory_update_type = 'bfgs';
-    options2.ipopt.print_level = 1;
-
-    %run fit and calculate actual sigma by squared substracting initial
-    %sigma / spotsize
-
-    start2 = [0.002, preSigma1, preSigma2];
-    [fitResult, ~] = ipopt (start2, funcs2, options2);
-    
-    preSigma2 = fitResult(3);
-    
+    % fitting for either matlab or octave_core_file_limit
+    if ~matRad_cfg.isOctave
+      
+        % first single gauss fit
+        funcs1.objective = @(p) sum((gauss1(axisGauss, p(1)) - profile).^2, 'all');
+        
+        funcs1.gradient  = @(p) 2 * sum((gauss1(axisGauss, p(1)) - profile) .* ...  
+                                    exp(-axisGauss.^2 ./ (2 * p(1)^2)) .* (axisGauss.^2 - 2 .* p(1)^2) ./ (2 .* pi .* p(1)^5), 'all');
+                                 
+                                         
+        options1.lb =   0;
+        options1.ub = Inf;
+        options1.ipopt.hessian_approximation = 'limited-memory';
+        options1.ipopt.limited_memory_update_type = 'bfgs';
+        options1.ipopt.print_level = 1;
+        start1 = 8;
+        
+        [fitResult1, ~] = ipopt (start1, funcs1, options1);                   
+        preSigma1 = fitResult1;   
+        
+        %define fit parameters
+        funcs2.objective = @(p) sum(sum((gauss2(axisGauss, p(1), p(2), p(3)) - profile).^2));
```

Comment:
If the function definitions are the same in the Matlab & Octave branch, maybe you can save the handles first and then make the if-clause. Now they are implemented twice, right?

Reply:
Here, the octave and matlab functions are not all precisely the same, as Octave does not allow to use the keyword "all" in the sum function. This is why I did not move the function declerations out of the if statement.